addons/mail/models/discuss/discuss_channel.py

```diff
@@ -763,10 +763,12 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
         self._bus_send("discuss.channel/new_message", payload)
         return rdata
 
-    def _notify_by_web_push_prepare_payload(self, message, msg_vals=False):
-        payload = super()._notify_by_web_push_prepare_payload(message, msg_vals=msg_vals)
+    def _notify_by_web_push_prepare_payload(self, message, msg_vals=False, force_record_name=False):
+        payload = super()._notify_by_web_push_prepare_payload(
+            message, msg_vals=msg_vals, force_record_name=force_record_name,
+        )
         payload['options']['data']['action'] = 'mail.action_discuss'
-        record_name = msg_vals['record_name'] if 'record_name' in msg_vals else message.record_name
+        record_name = force_record_name or message.record_name
```

Comment:
Note: we probably meant to just use the channel (display_)name here, I don't think we ever need/want to force it, but we can consider it out of scope, the PR as minimal diff is ok here.